### PR TITLE
[Merged by Bors] - Consolidate global cluster config

### PR DESCRIPTION
**Pull request by maltesander**
# Description

The new structure now looks like:
```
apiVersion: zookeeper.stackable.tech/v1alpha1
kind: ZookeeperCluster
metadata:
  name: simple-zk
spec:
  image:
    productVersion: 3.8.0
    stackableVersion: 0.8.0
  clusterConfig:
    authentication:
      - authenticationClass: zookeeper-tls-authentication-class (with provider tls) # String
    tls:
      serverSecretClass: tls
      quorumSecretClass: tls
    logging:
      vectorAggregatorConfigMapName: vector-aggregator-discovery
  servers:
....
```

fixes https://github.com/stackabletech/zookeeper-operator/issues/596

test: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/zookeeper-operator-it-custom/33/

## Review Checklist

- [ ] Code contains useful comments
- [ ] CRD change approved (or not applicable)
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
- [ ] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: rust/crd/src/lib.rs [hunk 1]
```diff
@@ -88,75 +88,119 @@ pub enum Error {
...
 #[serde(rename_all = "camelCase")]
 pub struct ZookeeperClusterSpec {
-    /// Emergency stop button, if `true` then all pods are stopped without affecting configuration (as setting `replicas` to `0` would)
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub stopped: Option<bool>,
-    /// Desired ZooKeeper version
+    /// Global ZooKeeper cluster configuration that applies to all role and role groups.
+    #[serde(default = "cluster_config_default")]
+    pub cluster_config: ZookeeperClusterConfig,
+    /// Desired ZooKeeper version.
```
Comment (sbernauer) [inline rust/crd/src/lib.rs:94]:
```suggestion
    /// ZooKeeper image to use.
```

File: rust/crd/src/lib.rs [hunk 2]
```diff
@@ -88,75 +88,119 @@ pub enum Error {
...
-    #[serde(default = "global_config_default")]
-    pub config: GlobalZookeeperConfig,
+    /// ZooKeeper server configuration.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub servers: Option<Role<ZookeeperConfigFragment>>,
-    /// Name of the Vector discovery ConfigMap.
-    /// It must contain the key `ADDRESS` with the address of the Vector aggregator.
+    /// Emergency stop button, if `true` then all pods are stopped without affecting configuration (as setting `replicas` to `0` would).
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub vector_aggregator_config_map_name: Option<String>,
```
Comment (sbernauer) [inline rust/crd/src/lib.rs:103]: Not entirely sure if `vector_aggregator_config_map_name` should be override-able at a role and role-group level.
Same goes for `logging`, as i can image that you want to increase the ROOT log-level for a certain role such as the namenode.
CC @siegfriedweber

**Review by sbernauer — COMMENTED** [review]
Just some comment, no complete review.

Any particular reason you did go with
```
    tls:
      server:
        secretClass: tls
      quorum:
        secretClass: tls
```
instead of
```
    tls:
      serverSecretClass: secret_class # Option<String>. *In general* defaults to "tls"
      internalSecretClass: secret_class # Option<String>. *In general* defaults to "tls"
```

**Comment by maltesander** [comment]
> Just some comment, no complete review.
> 
> Any particular reason you did go with
> 
> ```
>     tls:
>       server:
>         secretClass: tls
>       quorum:
>         secretClass: tls
> ```
> 
> instead of
> 
> ```
>     tls:
>       serverSecretClass: secret_class # Option<String>. *In general* defaults to "tls"
>       internalSecretClass: secret_class # Option<String>. *In general* defaults to "tls"
> ```



> Just some comment, no complete review.
> 
> Any particular reason you did go with
> 
> ```
>     tls:
>       server:
>         secretClass: tls
>       quorum:
>         secretClass: tls
> ```
> 
> instead of
> 
> ```
>     tls:
>       serverSecretClass: secret_class # Option<String>. *In general* defaults to "tls"
>       internalSecretClass: secret_class # Option<String>. *In general* defaults to "tls"
> ```

It basically was like this for server, i did the same for quorum. I thought about just using the String and like it better as well. Happy to change this.

**Review by razvan — APPROVED** [review]
lgtm

**Comment by maltesander** [comment]
test for latest changes: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/zookeeper-operator-it-custom/33/

**Review by razvan — APPROVED** [review]
lgtm

File: rust/crd/src/lib.rs (hunk 2 above)
Comment (sbernauer) [reply rust/crd/src/lib.rs:103]: Ok for me, not needed. For the record: The loglevel can be set for every role-group separately

**Comment by fhennig** [comment]
Can you test if the getting started guide still works?

**Comment by sbernauer** [comment]
For the record: Spoke to Malte and we agreed that the `authentication`´section needs to be changed to a `Vec<>` to match the description of the PR. We should include all braking changes in this PR to not have another breaking change shortly afterwards

**Comment by maltesander** [comment]
hopefully last test: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/zookeeper-operator-it-custom/35/

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**